[CI] AzureRepositoriesMeteringIT::testStatsAreStoredIntoAN.. failed on master

**Build scan**:https://gradle-enterprise.elastic.co/s/jhuha5mam3rtm

**Repro line**:
```./gradlew ':x-pack:plugin:repositories-metering-api:qa:azure:integTest' --tests org.elasticsearch.xpack.repositories.metering.azure.AzureRepositoriesMeteringIT.testStatsAreStoredIntoANewCounterInstanceAfterRepoConfigUpdate" -Dtests.seed=F180996FB19C0C99 -Dtests.security.manager=true -Dtests.locale=th-TH-u-nu-thai-x-lvariant-TH -Dtests.timezone=America/Fort_Nelson -Druntime.java=11```

**Reproduces locally?**: no

**Applicable branches**: master

**Failure history**: 7/7days, started 2020/09/08

**Failure excerpt**:
```
org.elasticsearch.xpack.repositories.metering.azure.AzureRepositoriesMeteringIT > testStatsAreStoredIntoANewCounterInstanceAfterRepoConfigUpdate FAILED
10:51:31     java.lang.AssertionError: 
10:51:31     Expected: is a value greater than <0L>
10:51:31          but: <0L> was equal to <0L>
10:51:31         at __randomizedtesting.SeedInfo.seed([F180996FB19C0C99:CA94289FD9FD4C0A]:0)
10:51:31         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
10:51:31         at org.junit.Assert.assertThat(Assert.java:956)
10:51:31         at org.junit.Assert.assertThat(Assert.java:923)
10:51:31         at org.elasticsearch.xpack.repositories.metering.AbstractRepositoriesMeteringAPIRestTestCase.assertRequestCountersAccountedForWrites(AbstractRepositoriesMeteringAPIRestTestCase.java:270)
10:51:31         at org.elasticsearch.xpack.repositories.metering.AbstractRepositoriesMeteringAPIRestTestCase.lambda$testStatsAreStoredIntoANewCounterInstanceAfterRepoConfigUpdate$5(AbstractRepositoriesMeteringAPIRestTestCase.java:165)
10:51:31         at org.elasticsearch.xpack.repositories.metering.AbstractRepositoriesMeteringAPIRestTestCase.snapshotAndRestoreIndex(AbstractRepositoriesMeteringAPIRestTestCase.java:253)
10:51:31         at org.elasticsearch.xpack.repositories.metering.AbstractRepositoriesMeteringAPIRestTestCase.testStatsAreStoredIntoANewCounterInstanceAfterRepoConfigUpdate(AbstractRepositoriesMeteringAPIRestTestCase.java:160)
10:51:31         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
10:51:31         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
10:51:31         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
10:51:31         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
10:51:31         at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
10:51:31         at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
10:51:31         at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
10:51:31         at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
10:51:31         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
10:51:31         at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
10:51:31         at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
10:51:31         at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
10:51:31         at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
10:51:31         at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
10:51:31         at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)

```

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
@@ -126,8 +126,9 @@ public class AzureBlobStore implements BlobStore {
             } else if (queryParams.contains("comp=blocklist")) {
                 stats.putBlockListOperations.incrementAndGet();
             } else {
-                // if a sas token is used, it's possible that the query params contain
-                // additional parameters unrelated to the upload type
+                // https://docs.microsoft.com/en-us/rest/api/storageservices/put-blob#uri-parameters
+                // The only URI parameter allowed for put-blob operation is "timeout", but if a sas token is used,
+                // it's possible that the URI parameters contain additional parameters unrelated to the upload type.
                 stats.putOperations.incrementAndGet();
             }
         });

--- a/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
+++ b/plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java
@@ -117,11 +117,7 @@ public class AzureBlobStore implements BlobStore {
         };
         this.uploadMetricsCollector = (httpURLConnection -> {
            assert httpURLConnection.getRequestMethod().equals("PUT");
-            String queryParams = httpURLConnection.getURL().getQuery();
-            if (queryParams == null) {
-                stats.putOperations.incrementAndGet();
-                return;
-            }
+            String queryParams = httpURLConnection.getURL().getQuery() == null ? "" : httpURLConnection.getURL().getQuery();
 
             // https://docs.microsoft.com/en-us/rest/api/storageservices/put-block
             // https://docs.microsoft.com/en-us/rest/api/storageservices/put-block-list
@@ -129,6 +125,10 @@ public class AzureBlobStore implements BlobStore {
                 stats.putBlockOperations.incrementAndGet();
             } else if (queryParams.contains("comp=blocklist")) {
                 stats.putBlockListOperations.incrementAndGet();
+            } else {
+                // if a sas token is used, it's possible that the query params contain
+                // additional parameters unrelated to the upload type
+                stats.putOperations.incrementAndGet();
             }
         });
     }

